tensorflow/c/eager/gradients_test.cc

```diff
@@ -175,135 +93,6 @@ Status IdentityNGradModel(AbstractContext* ctx,
-  tape->Watch(inputs[1]);  // Watch y.
-  std::vector<AbstractTensorHandle*> mul_outputs(1);
-  AbstractContextPtr tape_ctx(new TapeContext(ctx, tape, registry));
-  TF_RETURN_IF_ERROR(ops::Mul(tape_ctx.get(), inputs,
-                              absl::MakeSpan(mul_outputs),
-                              "Mul"));  // Compute x*y.
-  TF_RETURN_IF_ERROR(tape->ComputeGradient(ctx, /*targets=*/mul_outputs,
-                                           /*sources=*/inputs,
-                                           /*output_gradients=*/{}, outputs));
-  for (auto mul_output : mul_outputs) {
-    mul_output->Unref();
-  }
-  delete tape;
-  return Status::OK();
-}
-
-// Computes
-// y = log(1 + inputs[0])
-// return grad(y, {inputs[0]})
-Status Log1pGradModel(AbstractContext* ctx,
-                      absl::Span<AbstractTensorHandle* const> inputs,
-                      absl::Span<AbstractTensorHandle*> outputs) {
-  GradientRegistry registry;
-  TF_RETURN_IF_ERROR(RegisterGradients(&registry));
-  auto tape = new Tape(/*persistent=*/false);
-  tape->Watch(inputs[0]);  // Watch x.
-  std::vector<AbstractTensorHandle*> log1p_outputs(1);
-  AbstractContextPtr tape_ctx(new TapeContext(ctx, tape, registry));
-  TF_RETURN_IF_ERROR(ops::Log1p(tape_ctx.get(), inputs,
-                                absl::MakeSpan(log1p_outputs),
-                                "Log1p"));  // Compute log(1 + x).
-  TF_RETURN_IF_ERROR(tape->ComputeGradient(ctx, /*targets=*/log1p_outputs,
-                                           /*sources=*/inputs,
-                                           /*output_gradients=*/{}, outputs));
-  for (auto log1p_output : log1p_outputs) {
-    log1p_output->Unref();
-  }
-  delete tape;
-  return Status::OK();
-}
-
-// Computes
-// y = inputs[0] / inputs[1]
-// return grad(y, {inputs[0], inputs[1]})
-Status DivNoNanGradModel(AbstractContext* ctx,
-                         absl::Span<AbstractTensorHandle* const> inputs,
-                         absl::Span<AbstractTensorHandle*> outputs) {
-  GradientRegistry registry;
-  TF_RETURN_IF_ERROR(RegisterGradients(&registry));
-  auto tape = new Tape(/*persistent=*/false);
-  tape->Watch(inputs[0]);  // Watch x.
-  tape->Watch(inputs[1]);  // Watch y.
-  std::vector<AbstractTensorHandle*> div_outputs(1);
-  AbstractContextPtr tape_ctx(new TapeContext(ctx, tape, registry));
-  TF_RETURN_IF_ERROR(ops::DivNoNan(tape_ctx.get(), inputs,
-                                   absl::MakeSpan(div_outputs),
-                                   "DivNoNan"));  // Compute x / y.
-  TF_RETURN_IF_ERROR(tape->ComputeGradient(ctx, /*targets=*/div_outputs,
-                                           /*sources=*/inputs,
-                                           /*output_gradients=*/{}, outputs));
-  for (auto div_output : div_outputs) {
-    div_output->Unref();
-  }
-  delete tape;
-  return Status::OK();
-}
-
 Status getValue(AbstractTensorHandle* t, TF_Tensor** result_tensor) {
```

Comment:
Can we remove this and replace usage with tensorflow::gradients::GetValue?

Reply:
Done